### ISSUE TITLE
OpCode Bytes - Change from Vec to normal Array

### ISSUE THREAD

**Issue by Kevin27954** [Kevin27954/Pace_Honor_Thesis#12]
Due to Vec being stored on the heap, the speed of grabbing indexes is slower. It will start to add up over time especially the larger the number of opcodes we need to do (eg. loops, recursions).

Transition over to normal array and implement a reallocate function to increase the size of said array.

This is good since we cache the array (for some parts it's actually pretty minimal amount of opcode - simple programs usually less than 100), and should drastically improve the performance. 

Context:
In rust, accessing the same 5 u32 elements over and over again 300,000,000 times:
`Vec:    5532103 ms`
`Array:  2847996 ms`
Halve the time.

**Comment by Kevin27954** [comment]
This needs to be done in a similar manner to other areas that uses Vec. This way the speed should increase by quite an amount.

**Comment by Kevin27954** [comment]
This is good since we cache the array (for some parts it's actually pretty minimal amount of opcode), and should drastically improve the performance.

**Comment by Kevin27954** [comment]
This is impossible with the current state of code. It would require too much effort, which is basically a rewrite of the entire program. I would recommend starting from scratch for this if I were to do this issue. 

TLDR: For now just continue on and ignore the absolute trash speed of my code and accept it. We will change it in the future.

**Comment by Kevin27954** [comment]
Some how some way, it went from 60seconds for a fib(35) to a 35 second???

**Comment by Kevin27954** [comment]
I figure out how to make it faster. For now I will accept it, but next time I think I need to start from the ground up in order to make this compiler become even faster.
We can just let rust optimizer do it for us since in dev mode, there are a lot of checks and other code that Rust runs behind the scenes to ensure we get the maximum amount of safety when developing so when we build there will be no errors. And I had power saver mode on before, which basically reduced my CPU throughput by half. 
All in all, for fib(35), the best time I can do right now is 4secs after applying everything.
